DOC: Add a styleguide for developers. (#2230)
StackExchange/dnscontrol: documentation/styleguide.md

> # Coding Style

Reviewer: Make sure you include it within `SUMMARY.md`.

# Coding Style

# Use the Google Go Style Guide

https://google.github.io/styleguide/go/

# Always favor simplicity

This is a community project. The code you write today will be maintained by
someone in the future that may not be a professional developer or one that is
as experienced as you.

Remember what Brian Kernighan wrote: "Debugging is twice as hard as writing the
code in the first place. Therefore, if you write the code as cleverly as
possible, you are, by definition, not smart enough to debug it."

Remember the [John Woods](http://wiki.c2.com/?CodeForTheMaintainer) quote,
"Always code as if the guy who ends up maintaining your code will be a violent
psychopath who knows where you live."

Don't code for today-you.  Write code for six-months-from-now-you.  Have you

> met six-months-from-now-you? Oh, you should. Find individual. He or she is
> quite smart. He or she has 6 months more experience than today-you, but sadly

Reviewer 2: ```suggestion
met six-months-from-now-you? Oh, you should. Fine individual. They are
quite smart. They have 6 months more experience than today-you, but sadly
```
